kubejs/server_scripts/enigmatica/kubejs/expert/recipetypes/advancedperipherals/shaped.js

```diff
@@ -0,0 +1,116 @@
+onEvent('recipes', (event) => {
+    if (global.isExpertMode == false) {
+        return;
+    }
+    /*
+        ,
+        {
+            output: '',
+            pattern: ['', '', ''],
+            key: {
+                A: ''
+            },
+            id: ''
+        }
+    */
+
+    const recipes = [
+        {
+            output: 'advancedperipherals:peripheral_casing',
+            pattern: ['CDC', 'BAB', 'CEC'],
+            key: {
+                A: 'rftoolsbase:machine_frame',
+                B: 'create:redstone_link',
+                C: 'immersiveengineering:sheetmetal_iron',
+                D: 'integrateddynamics:part_world_reader',
+                E: 'integratedtunnels:part_player_simulator'
+            },
+            id: 'advancedperipherals:peripheral_casing'
+        },
+        {
+            output: 'advancedperipherals:chunk_controller',
+            pattern: ['BCB', 'DAE', 'BCB'],
+            key: {
+                A: 'advancedperipherals:peripheral_casing',
+                B: 'dustrial_decor:anchor',
+                C: 'minecraft:redstone_block',
+                D: 'minecraft:respawn_anchor',
+                E: 'mekanism:upgrade_anchor'
+            },
+            id: 'advancedperipherals:chunk_controller'
+        },
+        {
+            output: 'advancedperipherals:weak_automata_core',
+            pattern: ['ABA', 'CDC', 'AEA'],
+            key: {
+                A: 'emendatusenigmatica:gold_gear',
+                B: 'advancedperipherals:peripheral_casing',
+                C: 'tconstruct:soul_glass',
+                D: 'quark:soul_bead',
+                E: 'modularrouters:player_module'
+            },
+            id: 'advancedperipherals:weak_automata_core'
+        },
+        {
+            output: 'advancedperipherals:end_automata_core',
```

Comment:
Have you verified that this recipe works? :)

Reply:
Yes, this recipe works fine